### DISCUSSION TITLE
Request to extend the fuel gauge driver to include support for LC709203F

### DISCUSSION THREAD

**Discussion by zpm1066** [zephyrproject-rtos/zephyr#69305] [General] [answered]
**Is your feature request related to a problem? Please describe.**
Currently there's no driver for fuel gauge LC709203F. Some of the Adafruit ESP32 boards have LC709203F chip. Standalone LC709203F breakouts are also available to be used with a wide range of boards.

**Describe the solution you'd like**
I'd like to see the fuel gauge driver extended to include LC709203F. Currently, the driver supports MAX17048.

**Describe alternatives you've considered**
Purchasing new boards with MAX17048 fuel gauge is a possibility.

**Additional context**
Added LC709203F data sheet.

[LC709203F-D.PDF](https://github.com/zephyrproject-rtos/zephyr/files/14364043/LC709203F-D.PDF)

**Comment by henrikbrixandersen** [comment]
I can only encourage you to write a driver and submit a PR for this.

We cannot really take in requests or wishes for writing new drivers.

**Reply by zpm1066** [reply]
@henrikbrixandersen not an issue.
Any example on how to write a Zephyr driver? I’d be keen to do so. Thanks.

**Reply by henrikbrixandersen** [reply, marked as answer]
Yes! We have 1000+ examples of Zephyr drivers right here: https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers :-)
Examples of other Fuel Gauge drivers can be found here: https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/fuel_gauge

I'd suggest starting with reading https://docs.zephyrproject.org/latest/introduction/index.html and https://docs.zephyrproject.org/latest/kernel/drivers/index.html

**Reply by zpm1066** [reply]
@henrikbrixandersen Thanks.
I’ll review the existing two fuel gauge drivers and get familiar with the device driver model.
